feat(queue): allow to configure checks timeout (#3761)
Mergifyio/mergify-engine: docs/source/actions/queue.rst

       between 1 and 20.
       See :ref:`speculative checks`.

   * - ``checks_timeout``

>      - string or int

Reviewer: ```suggestion
     - string
```

We can convert it to int as needed 😉

>      - none

Reviewer: ```suggestion
     - 
```
We set it to empty when it's none.

>      - The times Mergify wait for Checks to succeed before automatically unqueue pull requests.
>        at least 60s. 50s, 5m or 1h can also be used for 1 second, 1 minute or 1 hour.

Reviewer: ```suggestion
     - The amonut of time Mergify wait for pending checks to return before unqueueing pull requests.
       This cannot be less than 60 seconds.
```

You need to document the format. It should be similar to the time ones we already used, though maybe they are not compatible since they used "ago".

Author: I hesitate on the format, this one is simple but doesn't look like the other.
The Postgres format may work if we remove the `ago` (eg: `2 days 18:02` or `18:02`). But it feels weird for a duration.
The ISO format is even worse: `P2DT18H02M0S`
If we keep this format do we support think like `1h 2m`?

Reviewer: I think you want to read https://www.postgresql.org/docs/9.1/datatype-datetime.html point 8.5.4 😄

Author: Yeah that the best one, we can't support Y and M has python timedelta doesn't support it

Author: § 8.5.4 support many formats:

Example | Description | Should we support it ?
-- | -- | --
1-2 | SQL standard format: 1 year 2 months | NO
3 4:05:06 | SQL standard format: 3 days 4 hours 5 minutes 6 seconds | MAYBE
1 year 2 months 3 days 4 hours 5 minutes 6 seconds | Traditional Postgres format: 1 year 2 months 3 days 4 hours 5 minutes 6 seconds | YES
P1Y2M3DT4H5M6S | ISO 8601 "format with designators": same meaning as above | NO
P0001-02-03T04:05:06 | ISO 8601 "alternative format": same meaning as above | NO

Reviewer: 100% agree. Your `maybe` is a `no` to me.
We actually say in the code IIRC we support Postgre format, not ISO nor SQL.
